src/userobjects/OpenMCDomainFilterEditor.C

```diff
@@ -0,0 +1,206 @@
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2024 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCDomainFilterEditor.h"
+#include "OpenMCProblemBase.h"
+#include "openmc/tallies/tally.h"
+#include "openmc/tallies/filter.h"
+#include "openmc/tallies/filter_cell.h"
+#include "openmc/tallies/filter_material.h"
+#include "openmc/tallies/filter_universe.h"
+#include "openmc/tallies/filter_mesh.h"
+
+registerMooseObject("CardinalApp", OpenMCDomainFilterEditor);
+
+InputParameters
+OpenMCDomainFilterEditor::validParams()
+{
+  InputParameters params = OpenMCUserObject::validParams();
+  params.addParam<bool>("create_filter", false, "Whether to create the tally if it doesn't exist");
+  params.addRequiredParam<int32_t>("filter_id", "The ID of the filter to modify");
+  params.addParam<MooseEnum>("filter_type", getFilterTypeEnum(), "The type of filter to create");
+  params.addRequiredParam<std::vector<std::string>>("bins", "The bins to modify in the filter");
+  params.declareControllable("bins");
+  params.addClassDescription("A UserObject for creating and managing OpenMC domain tally filters");
+  return params;
+}
+
+OpenMCDomainFilterEditor::OpenMCDomainFilterEditor(const InputParameters & parameters)
+  : OpenMCUserObject(parameters),
+    _filter_id(getParam<int32_t>("filter_id")),
+    _filter_type(getParam<MooseEnum>("filter_type").getEnum<OpenMCFilterType>())
+{
+
+  if (!parameters.isParamSetByUser("filter_type"))
+  {
+    mooseError(long_name() + ": filter_type must be specified");
+  }
+
+  this->initialize();
+}
+
+void
+OpenMCDomainFilterEditor::initialize()
+{
+  bool create_filter = getParam<bool>("create_filter");
+  bool filter_exists = this->filter_exists();
+
+  // if create_filter is set to true, but the filter already exists, display a warning
+  if (create_filter && filter_exists)
+  {
+    mooseWarning(long_name() + ": Filter " + std::to_string(_filter_id) +
```

Comment:
What is the intention for the user to understand when seeing this warning? I feel like this should be an error -- I guess the use case we are allowing is for a user to override an existing filter? 

Maybe we can add something to the error, like "You will be overwriting this filter."

Reply:
What I'd like them to know is that they specified that the filter should be created but that one with the spec provided is already present in the model to the `create_filter` was redundant. In that case there's nothing wrong (that I can see) with simply using that filter, but it seems useful for the user to know it was already there.